### PR TITLE
HL-67: API for employee data

### PR DESCRIPTION
**Pull request by haavikko**
## Description :sparkles:
* API for retrieving/editing employee data
* Validation and business rules will be added in later PR
* django admin for Employee

## Issues :bug:
HL-67

### CODE REVIEW

File: backend/benefit/applications/api/v1/serializers.py [hunk 1]
```diff
@@ -124,6 +128,80 @@ class Meta:
...
+                                                         10    A    26    U
+                                                         11    B    27    V
+                                                         12    C    28    W
+                                                         13    D    29    X
+                                                         14    E    30    Y
+                                                         15    F""".split()
+        )
+    }
+
+    def validate_social_security_number(self, value):
```
Comment (klemmari1) [inline backend/benefit/applications/api/v1/serializers.py:160]: Perhaps this validation could use some comments / docs on how it works :grin:

File: backend/benefit/requirements.in [hunk 2]
```diff
@@ -5,6 +5,7 @@ django-extensions
 djangorestframework
 django-simple-history
 psycopg2
+python-dateutil
```
Comment (klemmari1) [inline backend/benefit/requirements.in:8]: Missing from requirements.txt

File: backend/benefit/applications/api/v1/serializers.py [hunk 3]
```diff
@@ -350,25 +447,42 @@ def _validate_de_minimis_aid_set(self, company, de_minimis_aid, de_minimis_aid_s
...
+                    # employment and salary based benefits.
+                    # These two option have identical duration periods which need to be between 1-12 months.
+                    # (note: we'll allow full month ranges, like 2021-02-01 - 2021-02-28
+                    raise serializers.ValidationError(
+                        {"end_date": _("minimum duration of the benefit is one month")}
+                    )
+                if start_date + relativedelta(months=12) < end_date:
+                    raise serializers.ValidationError(
+                        {"end_date": _("minimum duration of the benefit is 12 months")}
+                    )
```
Comment (klemmari1) [inline backend/benefit/applications/api/v1/serializers.py:485]: I don't know the business logic here but based on
```
# These two option have identical duration periods which need to be between 1-12 months.
```

Should this be:
```
if start_date + relativedelta(months=12) > end_date:
    raise serializers.ValidationError(
        {"end_date": _("maximum duration of the benefit is 12 months")}
    )
```

File: backend/benefit/common/utils.py [hunk 4]
```diff
@@ -6,3 +6,21 @@ def update_object(obj, data):
             raise ValueError(f"{k} cannot be null.")
         setattr(obj, k, v)
     obj.save()
+
+
+def xgroup(iter, n=2, check_length=False):
```
Comment (klemmari1) [inline backend/benefit/common/utils.py:11]: Nice util function :+1:

File: backend/benefit/applications/api/v1/serializers.py (hunk 1 above)
Comment (haavikko) [reply backend/benefit/applications/api/v1/serializers.py:162]: Added link to the specification
I lifted this code with small modifications from a previous project (where I have full rights to the code).

File: backend/benefit/requirements.in (hunk 2 above)
Comment (haavikko) [reply backend/benefit/requirements.in:8]: added

File: backend/benefit/applications/api/v1/serializers.py (hunk 3 above)
Comment (haavikko) [reply backend/benefit/applications/api/v1/serializers.py]: Fixed the code and added tests
I'll be adding more validation logic in a future PR, this PR is needed in order to have something usable for the UI devs

**Review by klemmari1 — APPROVED** [review]
Looks good to me! :shipit:

**Pull request merged**